agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/immediately-house-card-abilities-resolution-game-state/mace-tyrell-ability-game-state/MaceTyrellAbilityGameState.ts

```diff
@@ -48,9 +48,15 @@ export default class MaceTyrellAbilityGameState extends GameState<
 
             this.parentGameState.onHouseCardResolutionFinish(house);
         } else {
+            const selectableFootmen = availableFootmen.every(fm => !fm.wounded) || availableFootmen.every(fm => fm.wounded)
+                // Automatically kill one of the footmen if all are wounded or all are not wounded
+                ? Array.of(availableFootmen[0])
+                // In case there are non-wounded and wounded footmen let the user decide which one to kill
+                : availableFootmen;
+
             this.setChildGameState(new SelectUnitsGameState(this)).firstStart(
                 house,
-                availableFootmen,
+                selectableFootmen,
```

Comment:
Is there a specific reason for calling `onSelectUnitsGameStateEnd` instead of setting a new childGameState? I'd prefer to do it that way, if possible.